pinot-core/src/main/java/com/linkedin/pinot/core/data/extractors/PlainFieldExtractor.java

```diff
@@ -124,7 +124,11 @@ public Schema getSchema() {
 
   @Override
   public GenericRow transform(GenericRow row) {
-    Map<String, Object> fieldMap = new HashMap<>();
+    return transform(row, new GenericRow());
+  }
+
+  @Override
+  public GenericRow transform(GenericRow row, GenericRow destinationRow) {
```

Comment:
The clear is done by the caller. It's assumed that if you give a destination row to write into, you already made sure that it is cleared (or that it is a brand new row to write into).